### PR TITLE
chore: Wording fixes for framework documents

### PR DESCRIPTION
**Pull request by acioc**
_Description of changes:_

Wording changes for documents under the framework directory.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: framework/raw-aes-keyring.md [hunk 1]
```diff
@@ -189,21 +189,21 @@ deserialize the [serialized key provider info](#key-provider-information) to obt
...
+- It uses the [authentication tag](#authentication-tag) obtained from deserialization as the AES-GCM input authentication tag.
+- It uses this keyring's [wrapping key](#wrapping-key) as the AES-GCM cipher key.
+- It uses the [IV](#iv) obtained from deserialization as the AES-GCM IV.
+- It uses the serialized [encryption context](structures.md#encryption-context-1) as the AES-GCM AAD.
 
 If a decryption succeeds, this keyring MUST
 add the resulting plaintext data key to the decryption materials and return the modified materials.
 
-If no decryption succeeds, the decryption MUST NOT make any update to the decryption materials.
+If no decryption succeeds, the decryption MUST NOT make any update to the decryption materials and MUST fail.
```
Comment (acioc) [inline framework/raw-aes-keyring.md:209]: Intentionally leaving vague so we can fix this under https://github.com/awslabs/aws-encryption-sdk-specification/issues/185

`MUST fail` wording provided by @mattsb42-aws

File: framework/raw-rsa-keyring.md [hunk 2]
```diff
@@ -140,20 +140,20 @@ The keyring MUST attempt to decrypt the inputted encrypted data keys, in list or
...
+- This keyring's [padding scheme](#supported-padding-schemes) is the RSA padding scheme.
+- An encrypted data key's [ciphertext](structures.md#ciphertext) is the input ciphertext to RSA decryption.
 
 If any decryption succeeds, this keyring MUST immediately return the input
 [decryption materials](structures.md#decryption-materials), modified in the following ways:
 
 - The output of RSA decryption is set as the decryption material's plaintext data key.
 
-If no decryption succeeds, this keyring MUST NOT make any update to the decryption materials.
+If no decryption succeeds, this keyring MUST NOT make any update to the decryption materials and MUST fail.
```
Comment (acioc) [inline framework/raw-rsa-keyring.md:159]: Intentionally leaving vague so we can fix this under https://github.com/awslabs/aws-encryption-sdk-specification/issues/185

`MUST fail` wording provided by @mattsb42-aws

File: framework/raw-aes-keyring.md [hunk 3]
```diff
@@ -117,10 +117,10 @@ the [ciphertext](structures.md#ciphertext) field in
...
 The bytes are appended in the order shown.
 
-| Field              | Length (bytes)                                          | Interpreted as |
-| ------------------ | ------------------------------------------------------- | -------------- |
-| Encrypted Key      | length of AES-GCM ciphertext output                     | Bytes          |
-| Authentication Tag | [Authentication Tag Length](#authentication-tag-length) | Bytes          |
+| Field              | Length (bytes)                                                   | Interpreted as |
+| ------------------ | ---------------------------------------------------------------- | -------------- |
+| Encrypted Key      | length of AES-GCM ciphertext output                              | Bytes          |
+| Authentication Tag | [Authentication Tag Length](#authentication-tag-length) as Bytes | Bytes          |
```
Comment (acioc) [inline framework/raw-aes-keyring.md:123]: Authentication Tag Length represents bits, not bytes. This needs to be converted for the table to make sense.

File: framework/caching-cmm.md [hunk 4]
```diff
@@ -144,7 +144,7 @@ The caching CMM MUST use the formulas specified in [Appendix A](#appendix-a-cach
 in order to compute the [cache entry identifier](cryptographic-materials-cache.md#cache-identifier).
 
 If a cache entry is found, the caching CMM MUST return the encryption materials retrieved.
-If a cache entry is not found, the caching CMM MUST then attempt to obtain the encryption materials
+If a cache entry is not found or the cache entry is expired, the caching CMM MUST then attempt to obtain the encryption materials
```
Comment (acioc) [inline framework/caching-cmm.md:147]: If the cache entry is found, but its hypothetically expired, this should still apply.

This matches the wording for `Decrypt Materials` on line 167 of this document.

File: framework/raw-rsa-keyring.md [hunk 5]
```diff
@@ -110,23 +110,23 @@ The keyring MUST NOT derive a public key from a specified [private key](#private
 OnEncrypt MUST take [encryption materials](structures.md#encryption-materials) as input.
 
 If the [encryption materials](structures.md#encryption-materials) do not contain a plaintext data key,
-on encrypt MUST generate a random plaintext data key and set it on the [encryption materials](structures.md#encryption-materials).
+OnEncrypt MUST generate a random plaintext data key and set it on the [encryption materials](structures.md#encryption-materials).
 
 The keyring MUST attempt to encrypt the plaintext data key in the
 [encryption materials](structures.md#encryption-materials) using RSA.
-The keyring performs RSA with the following specifics:
+The keyring performs RSA encryption with the following specifics:
```
Comment (lavaleri) [inline framework/raw-rsa-keyring.md:117]:
```suggestion
The keyring performs [RSA encryption](#rsa) with the following specifics:
```
We should link to our definition above here.

File: framework/raw-aes-keyring.md [hunk 6]
```diff
@@ -133,54 +133,54 @@ The authentication tag outputted from the AES-GCM encryption.
...
-[encryption context](structures.md#encryption-context-1) in the same format as the serialization of
-[message header AAD key value pairs](../data-format/message-header.md#key-value-pairs).
-If the keyring cannot serialize the encryption context, On Decrypt MUST fail.
+[encryption context](structures.md#encryption-context-1) in the same format
+as the serialization of the [message header AAD key value pairs](../data-format/message-header.md#key-value-pairs).
+If the keyring cannot serialize the encryption context, OnDecrypt MUST fail.
 
 The keyring MUST perform the following actions on each [encrypted data key](structures.md#encrypted-data-key)
-in the input encrypted data key list, serially, until it successfully decrypts one.
+in the inputted encrypted data key list, serially, until it successfully decrypts one.
```
[the remark this answers is not in the document]
Comment (acioc) [reply framework/raw-aes-keyring.md:183]: Updating.

File: framework/keyring-interface.md [hunk 7]
```diff
@@ -188,8 +188,8 @@ it returns on [OnEncrypt](#onencrypt) such that tampered versions of those encry
 if inputted into [OnDecrypt](#ondecrypt), are overwhelmingly likely to cause a decryption failure
 (i.e. the chance of a successful decryption in this case is negligible).
 
-Such integrity guarantees SHOULD include the integrity of the [encryption context](structures.md#encryption-context)
-such that, if the encryption context used as input to OnEncrypt to produce an encrypted data key is
+Such integrity guarantees SHOULD include the integrity of the [encryption context](structures.md#encryption-context).
+such that, if the encryption context is used as input to OnEncrypt to produce an encrypted data key that is
```
[the remark this answers is not in the document]
Comment (acioc) [reply framework/keyring-interface.md:192]: Yeah - I think I misunderstood this. Updating.

**Pull request merged**